### ISSUE TITLE
Issue with 3D files

### ISSUE THREAD

**Issue by marekvse** [ahmsville/Ahmsville-Dial-V2#1]
Hi Ahmed,
I have bought the full kit for Absolute variant from you, wanting to print my own colors. I first printed the case for the USB receiver, but it is obviously wrong. I have purchased the 3D files several weeks before purchasing the kit so the latest 3D files may have changed, but I don't want to purchase the same thing again just to find out if there is another version of the receiver case. In the version of 3D files I have there is v 1 and v1.1 of these. 
Is there a newer version? Could you send it to me please? Just for the receiver would be enough, unless something else changed, too.

And I have an additional question if you don't mind...

In the purchased 3D files there are gcode files to print the multi-color parts, but they are (I think obviously, though I'm not a 3D printing expert), not going to work for all 3D printers. I have Creality Ender 3 S1 Plus, which seems to be using a different flavour of GCode.  Would you have some instructions on how to prepare the gcode files from your STLs that are separate for the logo and the rest of the top button for example?

Thanks,
Marek

**Comment by ahmsville** [comment]
Hi marek,

Please note that the 3d files are free to download as many times as you like, gumroad only gives you the option to donate to the creator, all you need to do is enter 0 as the price and you can download the latest versions of the 3d files.

the included gcode files are outdated now so only use them as a reference on how to achieve the 2 color print.

**Comment by marekvse** [comment]
I see. Thanks for that! :) I downloaded the USB receiver files and sure enough the 1.2 version looks like that should fit the rest I have from you :).

Thank you for a very nice design, too. I have not put it together yet as I'm currently printing the parts, but it does look very well designed. I'm looking forward to experiencing it operate.

I still have no idea how to make any slicer to go over the 3 layers of the logo to avoid them when printing the rest of the knob. I only had partial success in printing it. It's got some holes where the logo parts were pushed out of position. If I printed just the first layer of the logo it would be fine, but would it be strong enough...?

**Comment by ahmsville** [comment]
you need at least two layers for the Surface, I could slice the parts for you if you'll like, I use a sapphire plus direct drive 0.6mm nozzle printer, not too far from what you have, so if you can tell me some settings specific to your printer, I can generate a useable gcode for you.

**Comment by marekvse** [comment]
That would be great!! :)
I have prepared 1 and 2 layer logo files to try after everything else is printed, but my problem is that I can't seem to be able to make the printing head go up between the places where it prints to not bump into the others. Also, without that and 3 layers the height sensor (BLTouch) is already bumping into the printed layers as well, even though it was still able (just) to stumble over them.
So if you have a facility that makes the Z jumps possible or have some other clever technique I'd be very very happy if you could prepare the gcode for me. Much appreciated!

So, I'm not sure what you may need for the specific settings exactly, but here is what I assume you may need. If anything is missing please let me know:

**Printer settings:**
width: 300mm
depth: 300mm
rectangular
G-code flavor: Marlin

**Printhead settings:**
X min: -26mm
Y min: -32mm
X max: 32 mm
Y max: 34mm
Gantry height: 25mm
Apply extruder offsets to GCode: Yes

I have 0.4mm nozzle on it.

**Start G-code:**
```
; Ender 3 S1 Start G-code
G92 E0 ; Reset Extruder
G28 ; Home all axes
G1 Z10.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0 Y0

M104 S{material_print_temperature_layer_0}
M190 S{material_bed_temperature_layer_0}
M109 S{material_print_temperature_layer_0}

G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
```

**End G-code:**
```
G91 ;Relative positioning
G1 E-2 F2700 ;Retract a bit
G1 E-2 Z0.2 F2400 ;Retract and raise Z
G1 X5 Y5 F3000 ;Wipe out
G1 Z10 ;Raise Z more
G90 ;Absolute positioning

G1 X0 Y{machine_depth} ;Present print
M106 S0 ;Turn-off fan
M104 S0 ;Turn-off hotend
M140 S0 ;Turn-off bed

M84 X Y E ;Disable all steppers but Z
```

**Comment by ahmsville** [comment]
the logo printing is done with Z hop, see the gcodes attached, print knob 1_1 first (logo), then print knob 1_2 after (knob), you can also preview the files before printing.
[KNOB 1 GCODE.zip](https://github.com/ahmsville/Ahmsville-Dial-V2/files/10691714/KNOB.1.GCODE.zip)

**Comment by marekvse** [comment]
Excellent, thank you so very much.

RE Z hop: Yes, I discovered that when looking for a solution, but it did not seem to work. Maybe the printer does not recognize that G-code...? I'll sure find out when I try the one from you.

**Comment by marekvse** [comment]
By the way, how do you make the keycaps fit? 
For others who may have ordered the kit, if that helps, I first tried a fine cutting instrument for a little bit, which was not working too well, before I tried directed hot air (using a soldering station) to heat up the cross-shaped hole for the switch a bit and then pressed it into the switch. It becomes a bit lose for a while but when it cools down again the fit is almost perfect (could be a tiny bit tighter, but its perfectly fine it seems, at least for something like this; I'm not sure I'd use this technique for a ftull keyboard, but maybe even that would work).

**Comment by ahmsville** [comment]
I don't have to do any post processing with the keys, the secret is to use the avoid traveling through holes settings, also print at the lower temperature range of the filament, you are also printing the keyed spacenav holder.

**Comment by marekvse** [comment]
I see.

Good o know if I want to change the color of the keys and thus print them again :).

Just to let you know that the G-code you made for me, unfortunately, did not work due to two problems. First, when it printed the logo it parked itself and did not finish the printing so that I could not select the second file to print. The second problem was something I should have let you know before, which is that my printer uses the BLTouch sensor at the beginning of every print and so I was doing my tries after I offset the printing position of the two files by 80mm on both x and y axis so that the center stays free for the BLTouch sensor step. Oh, actually the third problem was that it was (I think) printing the skirt and the retracted BLTouch tip was struggling to get across the already printed parts.

So in the end I tried by myself again, trying to print just 2 layers of the logo. That kind of worked, except that it dislodged in two of the four incomplete rings (I'm obviously printing the older design of the logo - just because I already had the gcode ready for it) and I then kind of fixed that with a 3D printing pen, filling manually the resulting inside ridges and thus also re-fixing the top to be all connected.
Thank you for the gcode files nevertheless!

Now I have two (probably last) questions if I may:

1) Will the capacitive touch work even after I made the inside top of the top knob slightly uneven after working with the hand-held 3D pen inside there? Is it designed so that it aligns perfectly with the inside top or is there a bit of space between the plastic and the capacitive sensor?

2) Does the dial work without having the battery installed?
  (I thought I don't need the wireless connection so I did not order the battery originally and it was only now that I realized that there is a possibility that the device will actually not work without it... So will it or will it not?)

Thanks,

Marek

**Comment by ahmsville** [comment]
Hi Merek,

The capacitive touch will work regardless of the top surface's smoothness, the dial will also work without a lipo battery, to bypass the lipo charging circuit you can desolder the x4 pads and connect the pads beside it, doing this will provide the dial with stable power in the absence of a lipo. see attached picture.
![20230214_125305_HDRl](https://user-images.githubusercontent.com/44074914/218736127-f626b161-ea46-4ab2-9e1c-8d56fcd0d870.jpg)

**Comment by marekvse** [comment]
Excellent. Thank you Ahmed for all your support. I still have not had the time yet to finish assembling the device but I can't wait to see it working.

**Comment by ahmsville** [comment]
I am happy to help.

**Comment by marekvse** [comment]
Hello Ahmed, sorry to bother you again. I have just finally found time to put all the magnets in and finish the assembly. In the meantime, the battery actually arrived so I connected the battery and thus powered the device for the first time. It seems to be running, the lights work, the capacitative touch works for Pause / Play youtube videos, but the thing I was most excited about to use it for of course was Fusion 360. I've followed the complete installation procedure and pressing the 5th button shows the popup identifying Fusion 360 correctly: 
![image](https://user-images.githubusercontent.com/11225208/224482592-df2a1940-3af7-420c-93b5-a0d4d940115c.png)
However, I'm getting no response whatsoever in Fusion from the dial so I cannot be sure that the dial actually works apart from the 5th button and the capacitative touch.
I have also confirmed that the Ahmsville Fusion 360 addin is installed and running in Fusion 360.

One thing I have not done yet is to push the knobs onto the bearings as it looks like it would be very hard, if not impossible, to get them off again after that and until I confirm that it all works I don't want to prevent myself from accessing the wiring.

Could you provide any guidance?

**Comment by marekvse** [comment]
Finally, I want to say thank you! :) I've finally managed to find the time to put all the magnets in, in the meantime, even the battery arrived and yesterday I finally managed to connect the device for the first time. Above, now hidden comment, I first wrote for help as only the capacitive touch seemed to work and no functions in Fusion 360, but then it finally, somehow, managed to start working. I found that I need a longer cable to use the device where I need it and I need to finish building my smaller keyboard to fit it all to the space I have, but the important thing is here that it works.

One question in the end... If I even need to disassemble it, have you tested whether the knobs can be taken off the bearings safely without breaking anything. I'd be worried if I ever needed to do that.

Oh, actually one more comment: The bottom knob rotates reasonably freely - that is if I give it a faster push it keeps rotating for about a second after the force ends (I expected better, but nevertheless it is very easy to rotate), but the top one has more resistance and cannot be easily rotated with just one finger without tilting the device. Is that by design to have to rotate it with a two-finger grip?
